File: NBE_5_7_2_02TEAM/src/main/java/io/twogether/nbe_5_7_2_02team/member/dto/response/LoginResponse.java
package io.twogether.nbe_5_7_2_02team.member.dto.response;

import io.twogether.nbe_5_7_2_02team.member.domain.Role;
import io.twogether.nbe_5_7_2_02team.oauth.dto.common.TokenPair;

import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@Builder
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class LoginResponse {
    public TokenPair tokenPair;

    public LoginResponse(TokenPair tokenPair, Role role, Long memberId) {
        this.tokenPair = tokenPair;
        this.role = role;
        this.memberId = memberId;
    }

    public Role role;
    public Long memberId;
}

File: NBE_5_7_2_02TEAM/src/main/java/io/twogether/nbe_5_7_2_02team/member/dto/response/LoginResponse.kt
package io.twogether.nbe_5_7_2_02team.member.dto.response

import io.swagger.v3.oas.annotations.media.Schema
import io.twogether.nbe_5_7_2_02team.member.domain.Role
import io.twogether.nbe_5_7_2_02team.oauth.dto.common.TokenPair

@Schema(description = "로그인 응답 DTO")
data class LoginResponse(
    @field:Schema(description = "액세스 및 리프레시 토큰 쌍")
    val tokenPair: TokenPair,

    @field:Schema(description = "회원 권한", example = "USER")
    val role: Role,

    @field:Schema(description = "회원 ID", example = "1")
    val memberId: Long,
)
